### ISSUE TITLE
Authorization isn't working properly

### ISSUE THREAD

**Issue by M7MDE7EGAZY** [FluentHub/FluentHub#138]
### Before you start...

- [X] Have you updated the FluentHub? You might be using an old version.
- [X] Have you checked if similar issue have already been reported?

### What windows are you seeing the problem on?

Windows 11 Dev

### What FluentHub version/architecture are you seeing the problem on?

0.3.0
X64

### Description

The Authorization isn't working

### Expected behavior

The Authorization must work to login

**Comment by onein528** [comment]
Can you paste log files or log screenshots?

**Comment by M7MDE7EGAZY** [comment]
`<img src="https://media.discordapp.net/attachments/957221676863602698/973903549136248872/Screenshot_2022-05-11_122612.png">
`

**Comment by M7MDE7EGAZY** [comment]
![Screenshot 2022-05-11 122612](https://user-images.githubusercontent.com/76889809/167838651-e0267ee6-da91-4bc8-bdd4-0a115bf5ff70.png)

**Comment by onein528** [comment]
can you rebuild with deleting all chaches(by solution cleaning or re-clone).

**Comment by M7MDE7EGAZY** [comment]
i'll try

**Comment by onein528** [comment]
on my machine, good work. so this is a not general issue.

**Comment by onein528** [comment]
@M7MDE7EGAZY Any progress?

**Comment by M7MDE7EGAZY** [comment]
nothing new , i'll try after exams again

**Comment by HelderRocket** [comment]
It also happens in Windows 10, the store version.
It does not advance from this screen.

![image](https://user-images.githubusercontent.com/104467221/169160014-9faebfab-a119-467d-8ff7-fd7fec406e94.png)

**Comment by onein528** [comment]
i can see same error on app which downloaded from ms store.

and error is Does not support Select(). in authorization phase

**Comment by onein528** [comment]
@M7MDE7EGAZY @HelderRocket Can you show me logs in FluentHub.Logs and FluentHub.Octokit.Logs?

**Comment by BobbyESP** [comment]
> ![Screenshot 2022-05-11 122612](https://user-images.githubusercontent.com/76889809/167838651-e0267ee6-da91-4bc8-bdd4-0a115bf5ff70.png)

I think this happened to us before @onein528

**Comment by onein528** [comment]
right. but suddenly fixed that at that time. at this time, suddenly appear this issue. very wierd. this is not our project's issue(maybe Octokit.NET issue).

**Comment by BobbyESP** [comment]
Yeah, maybe it's... When i finish my exams i would take a look

**Comment by onein528** [comment]
ok

**Comment by HelderRocket** [comment]
> @M7MDE7EGAZY @HelderRocket ¿Puede mostrarme registros en FluentHub.Logs y FluentHub.Octokit.Logs?

How do I do this?
😅😅

**Comment by onein528** [comment]
@M7MDE7EGAZY 

downloaded from Store: `C:\Users\username\AppData\Local\Packages\49462fluenthub-uwp.FluentHub_tq8wa33sfftnm\LocalState`

build from source: `C:\Users\username\AppData\Local\Packages\FluentHubAlpha_9es9q7t8fr9b8\LocalState`

**Comment by onein528** [comment]
Why is there difference.

![image](https://user-images.githubusercontent.com/62196528/169658135-130c4ea4-783e-4354-9701-1cfedddd6ddd.png)

**Comment by HelderRocket** [comment]
2022-05-18 16:31:38.572 -05:00 [ERR] RequestOAuthTokenAsync(): Select() is not supported
2022-05-18 16:33:06.703 -05:00 [INF] RequestGitHubIdentityAsync() completed successfully: [url: "https://github.com/login/oauth/authorize?client_id=0c0d30a8cbd9465373e2&scope=repo%2Cworkflow%2Cwrite%3Apackages%2Cdelete%3Apackages%2Cadmin%3Aorg%2Cadmin%3Apublic_key%2Cadmin%3Arepo_hook%2Cadmin%3Aorg_hook%2Cgist%2Cnotifications%2Cuser%2Cdelete_repo%2Cwrite%3Adiscussion%2Cadmin%3Aenterprise%2Cadmin%3Agpg_key"]
2022-05-18 16:33:09.561 -05:00 [INF] App.InitializeAsync() done
2022-05-18 16:33:09.961 -05:00 [ERR] RequestOAuthTokenAsync(): Select() is not supported
2022-05-18 16:34:37.754 -05:00 [ERR] RequestOAuthTokenAsync(): Select() is not supported
2022-05-18 16:35:38.569 -05:00 [INF] RequestGitHubIdentityAsync() completed successfully: [url: "https://github.com/login/oauth/authorize?client_id=0c0d30a8cbd9465373e2&scope=repo%2Cworkflow%2Cwrite%3Apackages%2Cdelete%3Apackages%2Cadmin%3Aorg%2Cadmin%3Apublic_key%2Cadmin%3Arepo_hook%2Cadmin%3Aorg_hook%2Cgist%2Cnotifications%2Cuser%2Cdelete_repo%2Cwrite%3Adiscussion%2Cadmin%3Aenterprise%2Cadmin%3Agpg_key"]
2022-05-18 16:35:39.887 -05:00 [INF] App.InitializeAsync() done
2022-05-18 16:35:40.317 -05:00 [ERR] RequestOAuthTokenAsync(): Select() is not supported
2022-05-18 16:35:49.322 -05:00 [ERR] RequestOAuthTokenAsync(): Select() is not supported

**Comment by onein528** [comment]
@HelderRocket From ms store?

**Comment by Mabc365** [comment]
same issue for me too

**Comment by HelderRocket** [comment]
Yes, it is the store version (I have the old version because I recently restored the computer).

![image](https://user-images.githubusercontent.com/104467221/169667444-5e8bca75-1938-4410-8864-96629a0b2087.png)

**Comment by onein528** [comment]
I will implement authorization service without using Octokit.Net. Also use row http request.

**Comment by onein528** [comment]
Then, i will publish app package, same as store version, can yall be tester?

@BobbyESP 
@M7MDE7EGAZY 
@Mabc365 
@HelderRocket 
@luandersonn 
@DeveloperWOW64

**Comment by DeveloperWOW64** [comment]
sure

**Comment by HelderRocket** [comment]
> Then, i will publish app package, same as store version, can yall be tester?
> 
> @BobbyESP @M7MDE7EGAZY @Mabc365 @HelderRocket @luandersonn @DeveloperWOW64

Yea, sure. 😃

**Comment by BobbyESP** [comment]
WHen i finish my exams i'll be tester and developer

**Comment by onein528** [comment]
Guys, I am ready to publish next generation v0.3.6 app package. Then I want you to test if authorization does work well.

To do for v0.3.6

- Update authorization phase
  - take a look at some docs for auth
- Update the structure of app sources(move methods in App.xaml.cs to other file and more).

If you encounter a issue, let me know what is wrong with app's log.

**Comment by DeveloperWOW64** [comment]
Via GH Releases?

**Comment by onein528** [comment]
No, via the artifacts which is generated by Build Azure Pipeline.

Give you the link later

**Comment by onein528** [comment]
Select used  in UserQueries.cs

First trying: (but it will fail)

https://fluenthub-community.visualstudio.com/FluentHub/_build/results?buildId=747&view=artifacts&pathAsName=false&type=publishedArtifacts

**Comment by onein528** [comment]
> Select used in UserQueries.cs

I can delete this code

**Comment by onein528** [comment]
Wow. unbelievable. FluentHub is terrible ans also fuck me up. i think it is better that I remove this project.

Cannot fix this issue.

**Comment by BobbyESP** [comment]
> Wow. unbelievable. FluentHub is terrible ans also fuck me up. i think it is better that I abandon this project.
> 
> Cannot fix this issue.

For me it works 
<img width="500" alt="image" src="https://user-images.githubusercontent.com/60316747/170737796-154e8f43-0115-4e68-ac6f-fce31381cff0.png">

**Comment by BobbyESP** [comment]
And well, if the people are not able of installing the app with the .msixbundle, they can just install it using the .ps1 file, is easier!
The .ps1 automatically installs the certificate in case you dont have it, and the app.

**Comment by BobbyESP** [comment]
https://user-images.githubusercontent.com/60316747/170740052-7820b598-d8a0-4ba8-878b-cce56dd8d32d.mp4

**Comment by DeveloperWOW64** [comment]
> And well, if the people are not able of installing the app with the .msixbundle, they can just install it using the .ps1 file, is easier! The .ps1 automatically installs the certificate in case you dont have it, and the app.

We can just code a Win32 that installs the ps1 and publish the win32 to the store

**Comment by HelderRocket** [comment]
> And well, if the people are not able of installing the app with the .msixbundle, they can just install it using the .ps1 file, is easier! The .ps1 automatically installs the certificate in case you dont have it, and the app.

Oh, yes, it worked like that.
Too bad it still doesn't advance from this screen.

![image](https://user-images.githubusercontent.com/104467221/170767271-f86be264-e589-4ceb-ad7e-48d1b777780b.png)

**Comment by DeveloperWOW64** [comment]
> Oh, yes, it worked like that. Too bad it still doesn't advance from this screen.

Sadly, there's no way to create a debug msixbundle. We'll just have to abandon upload to the Store. 😢

**Comment by BobbyESP** [comment]
Yeah :(( I'll take a look to it in a few hours if i'm able!

**Comment by onein528** [comment]
I understand. this is not auth error or issue. just Linq missing error i think.

**Comment by onein528** [comment]
it seems that auth method will fail when querying signed in user login name. I can delete this code. but if this doesn't work well, it means that all other queries will fail. So it's just a workaround and it can't be completely fixed and it seems to be due to the efficiency when generating the release package.

**Comment by DeveloperWOW64** [comment]
> it seems that auth method will fail when querying signed in user login name. I can delete this code. but if this doesn't work well, it means that all other queries will fail. So it's just a workaround and it can't be completely fixed and it seems to be due to the efficiency when generating the release package.

Create a seperate branch and experiment?

**Comment by onein528** [comment]
for what?

**Comment by DeveloperWOW64** [comment]
> I can delete this code. but if this doesn't work well, it means that all other queries will fail.

**Comment by onein528** [comment]
nice idea. i can.

**Comment by DeveloperWOW64** [comment]
> nice idea. i can.

It was your idea

**Comment by onein528** [comment]
More specific

```
2022-05-28 23:03:39.067 +09:00 [ERR] LoadHomeContentsAsync
System.NotSupportedException: Select() is not supported
   at Octokit.GraphQL.Core.Builders.QueryBuilder.RewriteListExtension(MethodCallExpression, MemberInfo) + 0x11e6
   at Octokit.GraphQL.Core.Builders.QueryBuilder.VisitMethodCall(MethodCallExpression, MemberInfo) + 0x9a
   at Octokit.GraphQL.Core.Builders.QueryBuilder.VisitMethodCall(MethodCallExpression) + 0xb
   at System.Linq.Expressions.MethodCallExpression.Accept(ExpressionVisitor) + 0x16
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression) + 0x18
   at Octokit.GraphQL.Core.Builders.QueryBuilder.Build[TResult](IQueryableList`1) + 0x5a
   at FluentHub.Octokit.Queries.Users.RepositoryQueries.<GetAllAsync>d__1.MoveNext() + 0x1d53
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17
   at FluentHub.ViewModels.Home.UserHomeViewModel.<LoadHomeContentsAsync>d__14.MoveNext() + 0x1a0

2022-05-28 23:03:42.918 +09:00 [ERR] MainPageViewModel.GetSignedInUser(): 
System.NotSupportedException: Select() is not supported
   at Octokit.GraphQL.Core.Builders.QueryBuilder.RewriteValueExtension(MethodCallExpression, MemberInfo) + 0x6b8
   at Octokit.GraphQL.Core.Builders.QueryBuilder.VisitMethodCall(MethodCallExpression, MemberInfo) + 0x56
   at Octokit.GraphQL.Core.Builders.QueryBuilder.VisitMethodCall(MethodCallExpression) + 0xb
   at System.Linq.Expressions.MethodCallExpression.Accept(ExpressionVisitor) + 0x16
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression) + 0x18
   at Octokit.GraphQL.Core.Builders.QueryBuilder.Build[TResult](IQueryableValue`1) + 0x40
   at FluentHub.Octokit.Queries.Users.UserQueries.<GetAsync>d__1.MoveNext() + 0xd66
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17
   at FluentHub.ViewModels.MainPageViewModel.<LoadSignedInUserAsync>d__75.MoveNext() + 0x142
```

**Comment by onein528** [comment]
@DeveloperWOW64 Experiment with GU10 latest commit `Experimental commit`

Artifacts here: https://fluenthub-community.visualstudio.com/FluentHub/_build/results?buildId=749&view=artifacts&pathAsName=false&type=publishedArtifacts

**Comment by HelderRocket** [comment]
It worked!
Well, in part.

![image](https://user-images.githubusercontent.com/104467221/170832655-ce41921d-4a1d-4be1-82d3-68c08e912aa8.png)

It does not load anything, not even by link.

**Comment by onein528** [comment]
yes, so do I. So this is not auth problem and also not FluentHub problem. just Linq problem. maybe missing assemblies for System.Linq.

**Comment by onein528** [comment]
I should notice. I decided to abandon the option to publish to ms store and github releases. I could not solve this issue. but some progress there is. this issue is not auth problem and is caused by Select() - one of the Linq extention methods - in Octokit.GraphQL.Net
It seems that this is general issue, and so not only our app.

I have another option to this. but I have no time to switch to that option. sorry for any inconvenience.

Exception:
```
Select() is not supported
```

**Comment by HelderRocket** [comment]
Is that error something generated by the tool you use to compile the application?

Sorry, I'm not very knowledgeable about how UWP apps work (although it even occurred to me to make a UWP app for Facebook and YT Music).

Maybe that bug can be reported to the tool developer.

**Comment by onein528** [comment]
Yes. I already reported to Octokit.GraphQL.net(is generated by this)

**Comment by onein528** [comment]
https://github.com/octokit/octokit.graphql.net/issues/262

but this project seems to be abandoned by owner(there are no activities these days)

**Comment by HelderRocket** [comment]
I see it's specifically geared towards Github libraries, so I don't think it's possible to do what I came up with.

**Comment by BobbyESP** [comment]
Maybe it's a good idea to mirror/fork the octokit library and fix the problem by ourselves

**Comment by onein528** [comment]
> what I came up with.

What is that.

**Comment by BobbyESP** [comment]
Well, forking the repository in the GitHub FluentHub organization and building us the dll files from source

**Comment by onein528** [comment]
the most problem is if we can fix that.

**Comment by BobbyESP** [comment]
Okey nice. And suddenly yeah, i think that is abandoned

**Comment by DeveloperWOW64** [comment]
It seems difficult - but it's worth a try

**Comment by BobbyESP** [comment]
Sorry, it's the only thing that occurs to me, just to try.

These days i'm gonna return to work. Today, my school is going to give me the final marks.

**Comment by onein528** [comment]
Hi all. @BobbyESP @M7MDE7EGAZY @Mabc365 @HelderRocket @luandersonn @DeveloperWOW64

Upon verification, there was no problem with authentication, and there was a problem with the code that was getting the user's information at the time of authentication. But this wasn't an exception caused by our code, it was caused by Octokit.GraphQL. So I [reported it to the GitHub Octokit.GraphQL team on GitHub](https://github.com/octokit/octokit.graphql.net/issues/262), but I think the project has been abandoned because I haven't received a reply for a week. We didn't want to use a non-maintainable project, and we came up with a new solution, but the cost of rewriting was huge and the code could be verbose. As a result, this project will not publish the release mode app package to the Microsoft Store or the release page on GitHub.

onein528 6/12/2022
